STYLE.md

```diff
@@ -131,15 +131,172 @@ if status is not pebble.ServiceStatus.ACTIVE:
+Abbreviations and acronyms in docstrings should usually be spelled out:
+
+- "for example" rather than "e.g."
+- "that is" rather than "i.e."
+- "and so on" rather than "etc"
+- "unit testing" rather than UT, and so on
 
 However, it's okay to use acronyms that are very well known in our domain, like HTTP or JSON or RPC.
+
+### Use sentence case in headings
+
+`## Use sentence case for headings`, instead of `## Use Title Case for Headings`.
+
+### Be consistent
+
+Choice of words: For example, if the whole document uses "mandatory", you probably shouldn't use "required" in a newly added paragraph. For another example, if the whole doc uses "list foo" when adding new content, don't use "get a list of bar".
+
+### Be precise
+
+Be precise in names and verbs. Use precise verbs to describe the behaviour. For example, the appropriate description of `/v1/services` is "list services", while "get a service" is probably a better fit for `/v1/services/{name}`.
+
+Split distinct ideas: Use separate sentences or clauses, and avoid cramming. Don’t merge unrelated details into a single phrase (for example, parameter format and default behaviour).
+
+Example:
+
+- Avoid: "The names of the services to get, a comma-separated string. If empty, get all services."
+- Prefer: "The names of the services to get. Specify multiple times for multiple values. If omitted, the method returns all services." (In three short precise sentences we've covered what the parameter does, usage details, and behavioral details.)
+
+Example:
+
+- Avoid: "For reference information about the API, see [API and clients] and [API]." (It's hard to tell the difference of these two.)
+- Prefer: "For an explanation of API access levels, see [API and clients]. For the full API reference, see [API reference]." (Clear)
+
+### Don't over-promote
+
+Avoid overstatement: Only use adjectives like "comprehensive", "powerful", or "robust" when the feature truly meets the description.
+
+Example:
+
+- Avoid: "Pebble provides a comprehensive health check feature"
+- Prefer: "Pebble provides a health check feature"
+
+Let features speak for themselves, describe actual capabilities and use measurable terms when possible (for example, "supports 3 types of checks" instead of "versatile checking")
+
+Future-proofing: Qualify with "currently" when describing evolving features. For example: "Currently, the only supported log target type is Grafana Loki".
+
+### Avoid implementation-specific terminology
+
+Focus on behaviour, not internal implementation: Describe what it does rather than how a particular implementation works.
+
+Example:
+
+- Avoid: "`Change.Err` will be non-empty if the change had an error." (Code-specific)
+- Prefer: "The `err` field in the response will contain an error message if the operation failed." (General)
```

Comment:
I feel that the problem here is that it's essentially a tautology, rather than being code specific. I think a better example in this space would be:

```suggestion
- Avoid: "The `Change` ID will be empty when the check is stopped." (Implementation-specific.)
- Prefer: "The `Status` will be 'inactive' when the check is stopped." (General.)
```